src/main/java/com/minecolonies/coremod/colony/buildings/BuildingTownHall.java

```diff
@@ -5,15 +5,21 @@
 import com.minecolonies.coremod.client.gui.WindowTownHall;
 import com.minecolonies.coremod.colony.Colony;
 import com.minecolonies.coremod.colony.ColonyView;
+import com.minecolonies.coremod.colony.permissions.PermissionEvent;
+import io.netty.buffer.ByteBuf;
 import net.minecraft.util.math.BlockPos;
 import org.jetbrains.annotations.NotNull;
 
+import java.util.LinkedList;
+import java.util.List;
+
+import static com.minecolonies.api.util.constant.ColonyConstants.MAX_PERMISSION_EVENTS;
 import static com.minecolonies.api.util.constant.ColonyConstants.NUM_ACHIEVEMENT_FIRST;
 
 /**
  * Class used to manage the townHall building block.
  */
-public class BuildingTownHall extends AbstractBuildingHut
+public class BuildingTownHall extends BuildingHome
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Override the "equals" method in this class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2160)